### PR TITLE
RavenDB-9492 fixing collection queries

### PR DESCRIPTION
**Pull request by talweiss1982**

### CODE REVIEW

File: src/Raven.Server/Documents/Queries/QueryFieldName.cs [hunk 1]
```diff
@@ -30,7 +30,7 @@ public QueryFieldName(string name, bool isQuoted)
 
         protected bool Equals(QueryFieldName other)
         {
-            return string.Equals(Value, other.Value, StringComparison.Ordinal) && IsQuoted == other.IsQuoted;
+            return string.Equals(Value, other.Value, StringComparison.OrdinalIgnoreCase) && IsQuoted == other.IsQuoted;
```
Comment (arekpalinski) [inline src/Raven.Server/Documents/Queries/QueryFieldName.cs:33]: Field names are case sensitive. Why did you need to change that?

File: src/Raven.Server/Documents/Queries/QueryMetadata.cs [hunk 2]
```diff
@@ -104,12 +104,25 @@ private void AddExistField(QueryFieldName fieldName, BlittableJsonReaderObject p
...
         }
 
-        public void AddWhereField(QueryFieldName fieldName, BlittableJsonReaderObject parameters, bool search = false, bool exact = false, AutoSpatialOptions spatial = null)
+        public void AddWhereField(QueryFieldName fieldName, 
+            BlittableJsonReaderObject parameters,            
+            bool search = false, 
+            bool exact = false, 
+            AutoSpatialOptions spatial = null,
+            OperatorType? operatorType = null,
+            bool isNegated = false
```
Comment (ayende) [inline src/Raven.Server/Documents/Queries/QueryMetadata.cs:113]: Why do you need this? The `operatorType` should be enough, no?

Comment (talweiss1982) [reply src/Raven.Server/Documents/Queries/QueryMetadata.cs:113]: no, it isn't, see:
var query = session.Query<Bunny>().Where(u => !u.Id.In(bunnies))
We get to id equal something but we lose the fact that we are under negation

**Comment by ayende** [comment]
rebased & merged
